src/smartMirrorProject/agendaWidget/views.py

```diff
@@ -1,3 +1,59 @@
 from django.shortcuts import render
+import os
+import requests
+from datetime import datetime
+from django.http import JsonResponse
 
-# Create your views here.
+# Debug API https://www.googleapis.com/calendar/v3/calendars/s4dsmartmirror@gmail.com/events?key=AIzaSyBlnu4fGIZeA3kuMZJEAMOtkh0JZSsyxRo
+
+AGENDA_KEY = os.getenv("AGENDA_KEY")
+
+# Helper function to format the date-time
+def format_datetime(datetime_str):
+    try:
+        # Fix malformed timezone offset by ensuring it ends with two digits
+        if datetime_str[-3] == ":" and len(datetime_str.split(":")[-1]) == 1:
+            datetime_str = datetime_str[:-1] + "0"  # Append a zero to fix offset
+
+        # Parse the datetime string with timezone info
+        dt_obj = datetime.strptime(datetime_str, "%Y-%m-%dT%H:%M:%S%z")
+        
+        # Format the datetime object into a human-readable string
+        return dt_obj.strftime('%d %b %Y, %H:%M')  # Example: 28 Nov 2024, 15:15
+    except Exception as e:
+        print(f"Error formatting datetime: {e} (input: {datetime_str})")
+        return datetime_str  # Fallback to the original string if parsing fails
+
+    
+# Fetch Google Calendar events
+def fetch_google_calendar_events(calendar_id):
+    url = f"https://www.googleapis.com/calendar/v3/calendars/{calendar_id}/events"
+    params = {
+        "key": AGENDA_KEY,  # Replace with your actual API key
```

Comment:
Als ik het goed begrijp is de constructie zo dat fetch_agenda_events_view het niet mogelijk is om de api key mee te geven als parameter, maar is het niet logischer om dit wel als parameter mee te geven in deze functie en pas in fetch_agenda_events_view als side effect in te laden.

Reply:
Dat lijkt me wel beter qua codekwaliteit en aanpasbaarheid. Heb dit toegepast door de AGENDA_KEY in te laden in fetch_agenda_events_view en mee te geven als hij fetch_google_calendar_events aanroept.